mitielib/mitie.py

```diff
@@ -555,4 +555,123 @@ def train(self):
+            # filename is a pointer to a ner object.
+            self.__obj = filename
+        else:
+            self.__obj = _f.mitie_load_text_categorizer(filename)
+        if (self.__obj == None):
+            raise Exception("Unable to load text_categorizer detector from " + filename)
+
+    def __del__(self):
+        self.__mitie_free(self.__obj)
+
+    def save_to_disk(self, filename):
+        """Save this object to disk.  You recall it from disk with the following Python
+        code: 
+            tcat = text_categorizer(filename)"""
+        if (_f.mitie_save_text_categorizer(filename, self.__obj) != 0):
+            raise Exception("Unable to save text_categorizer to the file " + filename);
+
+    
+    def __call__(self, tokens):
+        """Classify a relation object.  The input should have been produced by 
+        named_entity_extractor.extract_binary_relation().  This function returns a classification score
+        and if this number is > 0 then the relation detector is indicating that the input relation
+        is a true instance of the type of relation this object detects."""
+        score = ctypes.c_double()
+        label = ctypes.POINTER(ctypes.c_char_p)()
+        ctokens = python_to_mitie_str_array(tokens)
+        if (_f.mitie_categorize_text(self.__obj, ctokens, ctypes.byref(label), ctypes.byref(score)) != 0):
+            raise Exception("Unable to classify text.")
+        
+        label = ctypes.cast(label,ctypes.c_char_p)
+        return label.value, score.value 
+
+class text_categorizer_trainer(object):
+    def __init__(self, filename):
+        self.__obj = _f.mitie_create_text_categorizer_trainer(filename)
+        self.__mitie_free = _f.mitie_free
+        if (self.__obj == None):
+            raise Exception("Unable to create text_categorizer_trainer based on " + filename)
+
+    def __del__(self):
+        self.__mitie_free(self.__obj)
+
+    @property
+    def size(self):
+        return _f.mitie_text_categorizer_trainer_size(self.__obj)
+
+    def add_labeled_text(self, tokens, label):
+        ctokens = python_to_mitie_str_array(tokens)
+        if (_f.mitie_add_text_categorizer_labeled_text(self.__obj, ctokens, label) != 0):
+            raise Exception("Unable to add entity to training instance.  Probably ran out of RAM.");
+
+    @property
+    def beta(self):
+        return _f.mitie_text_categorizer_trainer_get_beta(self.__obj)
+
+    @beta.setter
+    def beta(self, value):
+        if (value < 0):
+            raise Exception("Invalid beta value given.  beta can't be negative.")
+        _f.mitie_text_categorizer_trainer_set_beta(self.__obj, value)
+
+    @property
+    def num_threads(self):
+        return _f.mitie_text_categorizer_trainer_get_num_threads(self.__obj)
+
+    @num_threads.setter
+    def num_threads(self, value):
+        _f.mitie_text_categorizer_trainer_set_num_threads(self.__obj, value)
+
+    def train(self):
+        #if (self.size == 0):
```

Comment:
Why is this commented out?  Shouldn't this check be here?

Also, the following comment is wrong.